### PR TITLE
catalog: clean up api reports

### PR DESCRIPTION
**Pull request by freben**
This is a really big one and I'd appreciate some extra eyes on it.

### CODE REVIEW

**Review by jhaals — COMMENTED** [review]
Great work! 🙌  There are a few previously unexported props that are really small which are now exported. Should we inline them to save on some exports? There are a few I commented on and some more.

File: plugins/catalog/src/components/AboutCard/AboutCard.tsx [hunk 1]
```diff
@@ -72,13 +72,20 @@ const useStyles = makeStyles({
   },
 });
 
-type AboutCardProps = {
+/**
+ * Props for {@link AboutCard}.
+ *
+ * @public
+ */
+export interface AboutCardProps {
```
[the remark this answers is not in the document]
Comment (freben) [reply plugins/catalog/src/components/AboutCard/AboutCard.tsx:80]: re: deprecated - wanted to do that across the board separately, it was in several places wasn't it 🤔 

re: inlining - hmm either that or lean toward consistency as a prio - not sure which is better

[the remark this answers is not in the document]
Comment (jhaals) [reply plugins/catalog/src/components/AboutCard/AboutCard.tsx:80]: consistency consistency consistency

**Review by jhaals — APPROVED** [review]

File: plugins/catalog/api-report.md [hunk 2]
```diff
@@ -98,275 +104,304 @@ const catalogPlugin: BackstagePlugin<
...
 // @public (undocumented)
-export const EntityHasSubcomponentsCard: ({
-  variant,
-}: {
-  variant?: 'gridItem' | undefined;
-}) => JSX.Element;
+export interface EntityLayoutProps {
+  // (undocumented)
+  children?: React_2.ReactNode;
+  // Warning: (ae-forgotten-export) The symbol "contextMenuOptions" needs to be exported by the entry point index.d.ts
```
Comment (Rugvip) [inline plugins/catalog/api-report.md:255]: some warnings here, but handle those later right?

**Review by Rugvip — APPROVED** [review]
HUUUUGE

File: plugins/catalog/api-report.md (hunk 2 above)
Comment (freben) [reply plugins/catalog/api-report.md:255]: yeah indeed - that's why i also didn't add it to the list in the api report generator script

**Pull request merged**